### ISSUE TITLE
Confusing message about making bucket.

### ISSUE THREAD

**Issue by deekoder** [minio/minio#4393]
```
$ mc mb mygcs mydeetestbucket
mc: <ERROR> Unable to make bucket `mygcs`. Bucket name cannot be empty
Bucket created successfully `mydeetestbucket`.

$ mc mb mygcs/mydeetestbucket
mc: <ERROR> Unable to make bucket `mygcs/mydeetestbucket`. The specified bucket is not valid.
```
Also confirmed that i have permissions to create bucket on gcs endpoint directly. Below is the error being shown on the minio server.

```
mc ls mygcs
mc: <ERROR> Unable to list folder. The specified bucket is not valid.
```
however mc cp works. Verified i have storage admin perms on gcs.

```
mc cp Downloads/videojs.imageOverlay.css mygcs/deefgsbucket
...erlay.css:  163 B / 163 B  ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  100.00% 201 B/s 0s
```

```
ERRO[11073] Unable to fetch bucket info.                  cause=Prefix access is denied: testbucket/ source=[gateway-handlers.go:830:gatewayAPIHandlers.HeadBucketHandler()] stack=~/go/src/github.com/minio/minio/cmd/gateway-gcs.go:243:(*gcsGateway).GetBucketInfo Users/Upheus1/go/src/github.com/minio/minio/cmd/gateway-handlers.go:824:(GatewayLayer).GetBucketInfo-fm ~/go/src/github.com/minio/minio/cmd/gateway-handlers.go:829:gatewayAPIHandlers.HeadBucketHandler ~/go/src/github.com/minio/minio/cmd/gateway-router.go:115:(gatewayAPIHandlers).HeadBucketHandler-fm
ERRO[11076] Unable to fetch bucket info.                  cause=Prefix access is denied: testbucket/ source=[gateway-handlers.go:830:gatewayAPIHandlers.HeadBucketHandler()] stack=~/go/src/github.com/minio/minio/cmd/gateway-gcs.go:243:(*gcsGateway).GetBucketInfo Users/Upheus1/go/src/github.com/minio/minio/cmd/gateway-handlers.go:824:(GatewayLayer).GetBucketInfo-fm ~/go/src/github.com/minio/minio/cmd/gateway-handlers.go:829:gatewayAPIHandlers.HeadBucketHandler ~/go/src/github.com/minio/minio/cmd/gateway-router.go:115:(gatewayAPIHandlers).HeadBucketHandler-fm
ERRO[11082] Unable to fetch bucket info.                  cause=Prefix access is denied: testbucket/ source=[gateway-handlers.go:830:gatewayAPIHandlers.HeadBucketHandler()] stack=~/go/src/github.com/minio/minio/cmd/gateway-gcs.go:243:(*gcsGateway).GetBucketInfo ~/go/src/github.com/minio/minio/cmd/gateway-handlers.go:824:(GatewayLayer).GetBucketInfo-fm ~/go/src/github.com/minio/minio/cmd/gateway-handlers.go:829:gatewayAPIHandlers.HeadBucketHandler ~/go/src/github.com/minio/minio/cmd/gateway-router.go:115:(gatewayAPIHandlers).HeadBucketHandler-fm
ERRO[11135] Unable to create a bucket.                    cause=Bucket name invalid: mydeetestbucket source=[gateway-handlers.go:673:gatewayAPIHandlers.PutBucketHandler()] stack=~/go/src/github.com/minio/minio/cmd/gateway-gcs.go:233:(*gcsGateway).MakeBucketWithLocation ~/go/src/github.com/minio/minio/cmd/gateway-handlers.go:671:gatewayAPIHandlers.PutBucketHandler ~/go/src/github.com/minio/minio/cmd/gateway-router.go:113:(gatewayAPIHandlers).PutBucketHandler-fm
```
## Expected Behavior
Clearer messaging on what the result of the operation is. 

## Current Behavior
 1. Error with success message is not helpful
 2. Unclear why the bucket is not valid.
Checked that bucket was never created in gcs.

## Steps to Reproduce (for bugs)
 1. Use the commands above to create a bucket.

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`): Darwin macOS Sierra
* Link to your project:

**Comment by harshavardhana** [comment]
> $ mc mb mygcs mydeetestbucket
> mc: <ERROR> Unable to make bucket `mygcs`. Bucket name cannot be empty
> Bucket created successfully `mydeetestbucket`.

`mc mb` is treating both as distinct arguments here.. Situations like this for example 

```
mc mb play/testbucket1 /tmp/testbucket1
Bucket created successfully `play/testbucket1`.
Bucket created successfully `/tmp/bucket1`.
```

Now `mc mb` works transparently on both i.e S3 and also filesystem. 

What happened in your scenario is `mygcs` was already an alias so it would show a message to be an error that you didn't provide a bucket name.  Second argument has no alias present so it would have created a directory locally as `mydeetestbucket` 

The syntax like `mc mb alias/<bucketname>` is needed to create bucket, you won't be able to create a bucket in this manner `mc mb alias <bucketname>`  - from a command line point of view, these are two distinct arguments one is point to S3 and another is filesystem (since there is no corresponding alias)

Say if you did this now 

```
$ mc mb play play
mc: <ERROR> Unable to make bucket `play`. Bucket name cannot be empty
mc: <ERROR> Unable to make bucket `play`. Bucket name cannot be empty
``` 

Notice that play is an alias.  

`mc` tool is designed to be transparent for both FS and S3. Say it doesn't differentiate i.e in-fact it makes FS look like S3 not the other way around.

**Comment by deekoder** [comment]
Root cause : `./minio gateway gcs projectid=projectgcstest1`
started gcs like this. It should error if this is not the right way to start gcs. Infact mc cp succeeds. Restarting as ` ./minio gateway gcs projectgcstest1` makes mc ls and mc mb work.
Thanks @poornas for finding the root cause.

**Comment by deekoder** [comment]
@harshavardhana I am not sure about what you say above. There are no issues with mc if i start gateway with  `MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio1234 ./minio gateway gcs projectgcstest1` but there is this issue when i start with `MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio1234 ./minio gateway gcs projectid=projectgcstest1`

**Comment by vadmeste** [comment]
This can be closed now.

**Comment by harshavardhana** [comment]
Closing as fixed in #4412
